# Descriptive Statistic & Visualization

## Descriptive Statistic

In [12]:
# Make Jupyter reload library before every execution

%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [28]:
import pandas as pd


df= pd.read_csv('data/all.csv', parse_dates=True, )
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 108 entries, 0 to 107
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Timestamp                        108 non-null    object 
 1   Age Group                        108 non-null    object 
 2   Gender                           108 non-null    object 
 3   Education Level                  108 non-null    object 
 4   Occupation                       108 non-null    object 
 5   Exercise Days/Week               108 non-null    object 
 6   Device Usage (hrs/day)           107 non-null    object 
 7   Screen Time Before Sleep         108 non-null    object 
 8   Height (cm)                      83 non-null     float64
 9   Weight (kg)                      92 non-null     float64
 10  Bedtime                          108 non-null    object 
 11  Wake-up Time                     108 non-null    object 
 12  Sleep Onset Time      

In [14]:
df.drop(labels=["Timestamp"], inplace=True, axis=1)
df.sample(5)

Age Group  Gender Education Level                  Occupation  \
73     25-34  Female        Master's                  Unemployed   
80     45-54  Female     High School                     Student   
40     25-34    Male       Doctorate  Professional/Office Worker   
43     35-44  Female      Bachelor's                     Service   
78     25-34  Female        Master's                       Other   

   Exercise Days/Week Device Usage (hrs/day) Screen Time Before Sleep  \
73           1-2 Days              4-6 Hours                 2+ Hours   
80           1-2 Days              4-6 Hours            30-60 Minutes   
40             0 Days               7+ Hours                 2+ Hours   
43           3-4 Days              4-6 Hours                1-2 Hours   
78             0 Days              1-3 Hours              <30 Minutes   

    Height (cm)  Weight (kg) Bedtime Wake-up Time Sleep Onset Time  \
73          NaN          NaN   02:00        09:00      >60 Minutes   
80          NaN         43.0   00:00        06:00    15-30 Minutes   
40        165.0          NaN   00:00        05:00    15-30 Minutes   
43        156.0         96.0   23:00        05:00    15-30 Minutes   
78          NaN          NaN   23:00        08:30    15-30 Minutes   

       Nap Duration Sleep Duration (hrs/24hr)  Sleep Quality  \
73           No Nap                 4-6 Hours              4   
80  Yes, 30-60 mins                 4-6 Hours              3   
40           No Nap                  6+ Hours              4   
43           No Nap                  6+ Hours              4   
78    Yes, <30 mins                 4-6 Hours              4   

   Sleep Disturbances Sleep Medication Language   BMI  \
73              Never               No  Bengali   NaN   
80              Never               No  Bengali   NaN   
40             Rarely               No  English   NaN   
43          Sometimes               No  English  39.4   
78          Sometimes               No  Bengali   NaN   

    Calculated Night Sleep Duration  
73                              7.0  
80                              6.0  
40                              5.0  
43                              6.0  
78                              9.5

In [25]:
# Calculating descriptive statistics for the dataset
descriptive_stats = df.describe()

# Since describe() includes only numeric data by default, adding additional descriptive statistics for non-numeric data
non_numeric_stats = df.describe(include=[object])


- **Sleep Quality:** On average, respondents rated their sleep quality around 3 on a scale, indicating moderate sleep quality.
- **BMI:** The average Body Mass Index (BMI) is around 23.55, with a range extending from 16.5 to 39.4.
- **Calculated Night Sleep Duration:** The average night sleep duration is around 7 hours, with a wide range from 1.67 hours to almost 9.75 hours.

In [26]:
descriptive_stats

Height (cm)  Weight (kg)  Sleep Quality        BMI  \
count    83.000000    92.000000     108.000000  80.000000   
mean    165.305542    67.415217       3.444444  24.552500   
std       8.321679    12.798085       0.824092   4.245503   
min     150.000000    43.000000       2.000000  17.500000   
25%     160.000000    59.800000       3.000000  21.500000   
50%     167.000000    68.000000       3.000000  23.550000   
75%     171.000000    75.000000       4.000000  26.600000   
max     185.000000   100.000000       5.000000  39.400000   

       Calculated Night Sleep Duration  
count                       105.000000  
mean                          7.036952  
std                           1.368431  
min                           1.670000  
25%                           6.500000  
50%                           7.000000  
75%                           8.000000  
max                           9.750000

- **Age Group:** The most common age group among respondents is 25-34.
- **Gender:** A slightly higher number of male respondents compared to females.
- **Education Level:** The majority of respondents have a Master's degree.
- **Occupation:** Many respondents are students.
- **Exercise Days/Week:** '1-2 Days' is the most common response for exercise frequency.
- **Device Usage (hrs/day):** A large portion of respondents use devices for '7+ Hours' per day.
- **Screen Time Before Sleep:** '30-60 Minutes' is the most common duration for screen time before sleep.
- **Sleep Disturbances:** 'Rarely' is the most frequent response, indicating that most respondents rarely experience sleep disturbances.
- **Sleep Medication:** The majority of respondents do not use sleep medication.
- **Language:** English is the most common language among respondents.

In [27]:
non_numeric_stats

Age Group Gender Education Level Occupation Exercise Days/Week  \
count        108    108             108        108                108   
unique         5      3               5          7                  4   
top        25-34   Male        Master's    Student           1-2 Days   
freq          72     67              47         47                 43   

       Device Usage (hrs/day) Screen Time Before Sleep Bedtime Wake-up Time  \
count                     107                      108     108          108   
unique                      3                        4      18           20   
top                  7+ Hours            30-60 Minutes   23:00        07:00   
freq                       43                       45      24           18   

       Sleep Onset Time Nap Duration Sleep Duration (hrs/24hr)  \
count               108          108                       107   
unique                4           11                         3   
top       15-30 Minutes       No Nap                  6+ Hours   
freq                 55           61                        64   

       Sleep Disturbances Sleep Medication Language  
count                 108              108      108  
unique                  5                2        4  
top                Rarely               No  English  
freq                   48              105       68

## Visualization

### Demographics

In [16]:
import plotly.express as px

fig = px.histogram(df, x="Age Group", color="Gender", title="Age Group over Gender Distribution")
fig.show()


In [17]:

fig = px.histogram(df, x="Education Level", color="Gender", title="Education Level over Gender Distribution")
fig.show()


In [18]:

fig = px.histogram(df, x="Occupation", color="Gender",title="Education Level over Gender Distribution")
fig.show()


### Behavior

In [19]:
for col in df.columns[5:8]:
    fig = px.histogram(df, x=col, color="Sleep Quality",title=f"{col} over Sleep Quality Distribution", histnorm="percent")
    fig.show()
